### PR TITLE
[ADD] solution hw-6.2

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,59 @@ public Integer getLength() {
...
+                && (getLength() == null && rectangle.getLength() == null)
+                && (getColor() == null && rectangle.getColor() == null)) {
+            return true;
+        }
+
+        if (getWidth().equals(rectangle.getWidth())
+                && getLength().equals(rectangle.getLength())
+                && getColor().equals(rectangle.getColor())) {
+            return true;
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: what if `width` is null and `rectangle.width` is not null?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,59 @@ public Integer getLength() {
...
+    @Override
+    public int hashCode() {
+        final int prime = 17;
+        int result = 31;
+
+        if (getColor() == null) {
+            result = prime * result;
+        } else {
+            result = prime * result + getColor().hashCode();
+        }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: try refactoring it to one line of code

also, do you need to use getters?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,44 @@ public Integer getLength() {
...
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        if ((getWidth() == null && rectangle.getWidth() == null)
+                || (getWidth() != null && getWidth().equals(rectangle.getWidth()))
+                && (getLength() == null && rectangle.getLength() == null)
+                || (getLength() != null && getLength().equals(rectangle.getLength()))
+                && (getColor() == null && rectangle.getColor() == null)
+                || (getColor() != null && getColor().equals(rectangle.getColor()))) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: why are you using getters inside the same class?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -27,4 +27,44 @@ public Integer getLength() {
...
+            return true;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]: don't overcomplicate if-else statements

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -27,4 +27,44 @@ public Integer getLength() {
...
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 17;
+        int result = 31;
+
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java:58]: there are too many blank lines in this method

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 5 above)
Comment (ghost) [reply src/main/java/core/basesyntax/Rectangle.java:58]: done

File: src/main/java/core/basesyntax/Rectangle.java [hunk 6]
```diff
@@ -27,4 +27,41 @@ public Integer getLength() {
...
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        if (width == rectangle.width
+                || (width != null && width.equals(rectangle.width))
+                && (length == rectangle.length)
+                || (length != null && length.equals(rectangle.length))
+                && (color == rectangle.color)
+                || (color != null && color.equals(rectangle.color))) {
+            return true;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
return (width == rectangle.width 
                || width != null && width.equals(rectangle.width))
                && (length == rectangle.length 
                || length != null && length.equals(rectangle.length))
                && (color == rectangle.color 
                || color != null && color.equals(rectangle.color));
```

Comment (KseniiaMakarova) [reply src/main/java/core/basesyntax/Rectangle.java]: The logical-AND operator ( && ) has higher precedence than the logical-OR operator ( || )
add parenthesis to this statement to ensure the flow is correct

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Rectangle.java (hunk 6 above)
Comment (ghost) [reply src/main/java/core/basesyntax/Rectangle.java]: done

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**